hazelcast-client-new/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java

```diff
@@ -69,6 +69,9 @@ public SerializationService createSerializationService() {
             SerializationServiceBuilder builder = new DefaultSerializationServiceBuilder();
             SerializationConfig serializationConfig = config.getSerializationConfig() != null ? config
                     .getSerializationConfig() : new SerializationConfig();
+            if (version > 0) {
```

Comment:
What if version <= 0 ? Unfortunately bytes in java are signed, so -1 is a possibility.